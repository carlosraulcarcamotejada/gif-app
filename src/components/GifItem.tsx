import { motion } from 'framer-motion';
import { FC } from 'react'

type props = {
    title:string,
    img:string,
}

const GifItem:FC<props> = ({title='',img}):JSX.Element => {

  // console.log(title);
  // console.log(img);

  return (
    <motion.div
    initial={{opacity:0}}
    animate={{opacity:1}}
    transition={{duration:0.4}}
    className='h-44 w-44 rounded-md shadow-md mb-8'>
        <h5 className='truncate text-white'>{title}</h5>
        <img className='h-44 w-44 object-cover rounded-b-md' src={img} alt={title} />
    </motion.div>
  )
}

export default GifItem;