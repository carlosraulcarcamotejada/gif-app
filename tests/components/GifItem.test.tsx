import { render,screen } from "@testing-library/react";
import React from "react";
import GifItem from '../../src/components/GifItem';



describe('Tests on <GifItem />', () => { 
    
    const props = {
        title:'one punch man',
        img:'https://www.giphy.com/anime/one-punch-man.gif',
    }


    test('Snapshot on <GifItem />', () => { 
        const {container} = render(<GifItem {...props} />)
        expect(container).toMatchSnapshot()
     })

    test('should show an image', () => { 
        render(<GifItem {...props}/>);
        const {src, alt} = screen.getByRole<HTMLImageElement>('img');
        expect(alt).toBe(props.title);
        expect(src).toBe(props.img);
    })
 
 })