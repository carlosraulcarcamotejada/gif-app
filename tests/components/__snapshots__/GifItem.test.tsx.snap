// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifItem /> Snapshot on <GifItem /> 1`] = `
<div>
  <div
    class="h-44 w-44 rounded-md shadow-md mb-8"
    style="opacity: 0;"
  >
    <h5
      class="truncate text-white"
    >
      one punch man
    </h5>
    <img
      alt="one punch man"
      class="h-44 w-44 object-cover rounded-b-md"
      src="https://www.giphy.com/anime/one-punch-man.gif"
    />
  </div>
</div>
`;
